p2p_invalid_block occasionally fails "bad-txns-inputs-duplicate"

Short log:

```
p2p_invalid_block.py failed, Duration: 1 s
stdout:
2018-11-05T15:27:36.459000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20181105_152228/p2p_invalid_block_46
2018-11-05T15:27:36.778000Z TestFramework (INFO): Create a new block with an anyone-can-spend coinbase
2018-11-05T15:27:36.890000Z TestFramework (INFO): Mature the block.
2018-11-05T15:27:37.238000Z TestFramework (INFO): Test merkle root malleability.
2018-11-05T15:27:37.301000Z TestFramework (INFO): Test duplicate input block.
2018-11-05T15:27:37.354000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/test/functional/test_framework/test_framework.py", line 171, in main
    self.run_test()
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/test/functional/p2p_invalid_block.py", line 92, in run_test
    node.p2p.send_blocks_and_test([block2_orig], node, success=False, request_block=False, reject_reason='bad-txns-inputs-duplicate')
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/test/functional/test_framework/mininode.py", line 543, in send_blocks_and_test
    assert node.getbestblockhash() != blocks[-1].hash
  File "/usr/lib/python3.5/contextlib.py", line 66, in __exit__
    next(self.gen)
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/test/functional/test_framework/test_node.py", line 272, in assert_debug_log
    self._raise_assertion_error('Expected message "{}" does not partially match log:\n\n{}\n\n'.format(expected_msg, print_log))
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/test/functional/test_framework/test_node.py", line 124, in _raise_assertion_error
    raise AssertionError(self._node_msg(msg))
AssertionError: [node 0] Expected message "bad-txns-inputs-duplicate" does not partially match log:
 - 2018-11-05T15:27:37.312967Z received: headers (82 bytes) peer=0
 - 2018-11-05T15:27:37.313141Z Requesting block 0b1e6087126df7c86349cec8019ee7134f485c79ad89e9ed180325c5b73cd878 from  peer=0
 - 2018-11-05T15:27:37.313156Z sending getdata (37 bytes) peer=0
 - 2018-11-05T15:27:37.316649Z received: ping (8 bytes) peer=0
 - 2018-11-05T15:27:37.316675Z sending pong (8 bytes) peer=0
 - 2018-11-05T15:27:37.318279Z received: block (327 bytes) peer=0
 - 2018-11-05T15:27:37.318321Z received block 0b4699cec0641f2d47013551446930cf165dfbcde6ff4440695e0678dd417068 peer=0
 - 2018-11-05T15:27:37.318358Z Misbehaving: 127.0.0.1:51816 peer=0 (3100 -> 3200)
 - 2018-11-05T15:27:37.318372Z ERROR: ProcessNewBlock: AcceptBlock FAILED (bad-txns-duplicate, duplicate transaction (code 16))
 - 2018-11-05T15:27:37.353006Z Received a POST request for / from 127.0.0.1:34964
 - 2018-11-05T15:27:37.353084Z ThreadRPCServer method=getbestblockhash user=__cookie__
```

Full log: https://travis-ci.org/bitcoin/bitcoin/jobs/450901528

--- a/test/functional/feature_block.py
+++ b/test/functional/feature_block.py
@@ -824,7 +824,7 @@ class FullBlockTest(BitcoinTestFramework):
         tx.vin.append(CTxIn(COutPoint(b64a.vtx[1].sha256, 0)))
         b64a = self.update_block("64a", [tx])
         assert_equal(len(b64a.serialize()), MAX_BLOCK_BASE_SIZE + 8)
-        self.sync_blocks([b64a], success=False, reject_reason='non-canonical ReadCompactSize():')
+        self.sync_blocks([b64a], success=False, reject_reason='non-canonical ReadCompactSize()')
 
         # bitcoind doesn't disconnect us for sending a bloated block, but if we subsequently
         # resend the header message, it won't send us the getdata message again. Just

--- a/test/functional/p2p_invalid_block.py
+++ b/test/functional/p2p_invalid_block.py
@@ -77,9 +77,9 @@ class InvalidBlockRequestTest(BitcoinTestFramework):
         block2.vtx.append(tx2)
         assert_equal(block2.hashMerkleRoot, block2.calc_merkle_root())
         assert_equal(orig_hash, block2.rehash())
-        assert(block2_orig.vtx != block2.vtx)
+        assert block2_orig.vtx != block2.vtx
 
-        node.p2p.send_blocks_and_test([block2], node, success=False, request_block=False, reject_reason='bad-txns-duplicate')
+        node.p2p.send_blocks_and_test([block2], node, success=False, reject_reason='bad-txns-duplicate')
 
         # Check transactions for duplicate inputs
         self.log.info("Test duplicate input block.")
@@ -89,7 +89,7 @@ class InvalidBlockRequestTest(BitcoinTestFramework):
         block2_orig.hashMerkleRoot = block2_orig.calc_merkle_root()
         block2_orig.rehash()
         block2_orig.solve()
-        node.p2p.send_blocks_and_test([block2_orig], node, success=False, request_block=False, reject_reason='bad-txns-inputs-duplicate')
+        node.p2p.send_blocks_and_test([block2_orig], node, success=False, reject_reason='bad-txns-inputs-duplicate')
 
         self.log.info("Test very broken block.")
 
@@ -102,7 +102,8 @@ class InvalidBlockRequestTest(BitcoinTestFramework):
         block3.rehash()
         block3.solve()
 
-        node.p2p.send_blocks_and_test([block3], node, success=False, request_block=False, reject_reason='bad-cb-amount')
+        node.p2p.send_blocks_and_test([block3], node, success=False, reject_reason='bad-cb-amount')
+
 
 if __name__ == '__main__':
     InvalidBlockRequestTest().main()

